lib/spack/llnl/util/filesystem.py

```diff
@@ -249,12 +249,25 @@ def install(src, dest):
     copy_mode(src, dest)
 
 
-def install_tree(src, dest, **kwargs):
-    """Manually install a directory tree to a particular location."""
+def install_tree(src, dest, symlinks=False):
+    """Manually install a directory tree to a particular location.
+
+    Arguments:
+        symlinks (bool): preserve symlinks if True (default False)
+    """
     tty.debug("Installing %s to %s" % (src, dest))
-    shutil.copytree(src, dest, **kwargs)
 
-    for s, d in traverse_tree(src, dest, follow_nonexisting=False):
+    for s, d in traverse_tree(src, dest, order='pre', follow_nonexisting=True):
+        if symlinks and os.path.islink(s):
+            # note that this won't rewrite absolute links into the old
+            # root to point at the new root. Should we handle that case?
```

Comment:
am I reading this correctly that e.g. `python -> oldabspath/python2.7` would still point to `oldabspath/python2.7` on the new location? If so, then yes we need to take care of this (albeit one could reasonably consider such packages broken anyways...)

not sure if it needs to be dealt with here, but at least a warning should be issued.